sos/report/plugins/ssh.py

```diff
@@ -33,7 +34,8 @@ def setup(self):
 
         sshcfgs = [
             "/etc/ssh/ssh_config",
-            "/etc/ssh/sshd_config"
+            "/etc/ssh/sshd_config",
+            "/etc/ssh/sshd_config.d/*",
             ]
```

Comment:
I ran an "old" sos (4.6) and it seems that we capture the .d directories already:
```
# ls /var/tmp/sosreport-localhost-2024-04-15-rxpnjai/etc/ssh/
ssh_config  ssh_config.d  sshd_config  sshd_config.d
```

And in the logs:

```
2024-04-15 09:58:56,424 INFO: [plugin:ssh] collecting path '/etc/ssh/sshd_config.d/50-redhat.conf'
2024-04-15 09:58:56,424 DEBUG: [plugin:ssh] copying path '/etc/ssh/sshd_config.d/50-redhat.conf' to archive:'/etc/ssh/sshd_config.d/50-redhat.conf'
2024-04-15 09:58:56,425 DEBUG: [archive:TarFileArchive] Making leading paths for /etc/ssh/sshd_config.d
```

Could be that when we run through sshcfgs below we take these as the beginning of the strings to check?

Reply:
Interesting.. Well, the `50-redhat.conf` is collected due to:

    Include /etc/ssh/sshd_config.d/*.conf

in `/etc/ssh/sshd_config`. Anyway I tested the file is honoured by `sshd` even without the `Include` line (and e.g. `https://manpages.debian.org/unstable/openssh-server/sshd_config.5.en.html` does suggest so as well).

Due to that "default include", it makes sense to dollect the dir (or just `*.conf` files..?), I think.